src/main/kotlin/systems/ajax/malov/stockanalyzer/service/impl/StockRecordAnalyzerServiceImpl.kt

```diff
@@ -26,7 +27,7 @@ class StockRecordAnalyzerServiceImpl(
             )
     }
 
-    override fun getAllManageableStocksSymbols(): List<String> {
-        return stockRecordRepository.findAllStockSymbols()
+    override fun getAllManageableStocksSymbols(): Mono<List<String>> {
+        return stockRecordRepository.findAllStockSymbols().collectList()
```

Comment:
should we return Flux<String> ?
we don't have processing of the whole list here

Reply:
https://github.com/ajax-malov-n/stockanalyzer/pull/8#discussion_r1771549939
I've got such a comment and changed my code. Or I need to make in a service Flux<String>, but on a controller level return Mono<List<String>>?

Reply:
You can change controller return type from `Mono<List<String>>` to `Flux<String>`.

Collecting items manually doesn't make any sense in this case